src/pages/search/Search.tsx

```diff
@@ -0,0 +1,169 @@
+      setSearchTerm("")
+    }
+
+    return
+  }
+
+  const renderSearchResults = () => {
+    const needle = searchTerm.toLocaleLowerCase();
+
+    const cases = data.cases.filter((x: any) => x.title.toLowerCase().includes(needle) || (x.summary ? x.summary.includes(needle) : null));
+    const topics = data.topics.filter((x: any) => x.title.toLowerCase().includes(needle) || (x.summary ? x.summary.includes(needle) : null));
+    const provisions = searchableProvisions.filter((x: any) => x.titleLower.includes(needle) || x.content.includes(needle));
+
+    return <>
+      <IonToolbar>
+        <IonSegment onIonChange={(e) => setCurrentSegment(e.detail.value || "all")} value={currentSegement}>
+          <IonSegmentButton value="all">
+            All
+          </IonSegmentButton>
+          <IonSegmentButton value="constitution">
+            Constitution
+          </IonSegmentButton>
+          <IonSegmentButton value="cases">
+            Cases
+          </IonSegmentButton>
+          <IonSegmentButton value="topics">
+            Topics
+          </IonSegmentButton>
+        </IonSegment>
+      </IonToolbar>
+      <IonList>
+        {renderProvisions(provisions)}
+        {renderCases(cases)}
+        {renderTopics(topics)}
+      </IonList>
+    </>
+  }
+
+  const renderProvisions = (provisions: any) => {
+    if (currentSegement === "all" || currentSegement === "constitution") {
+      return provisions.map((provision: any, index: any) => (
+        <IonItem key={index} routerLink={"/constitution/" + provision.id}>
+          <IonLabel>
+            <h3>{provision.title}</h3>
+          </IonLabel>
+        </IonItem>
+      ))
+    }
+  }
+
+  const renderCases = (cases: any) => {
+    if (currentSegement === "all" || currentSegement === "cases") {
+
+      return cases.map((article: any, index: any) => (
+        <IonItem key={index} routerLink={"/cases/" + article.id}>
+          <IonLabel>
+            <h3>{article.title}</h3>
+            <p>{parse(article.snippet)}</p>
+          </IonLabel>
+        </IonItem>
+      ))
+    }
+  }
+  const renderTopics = (topics: any) => {
+    if (currentSegement === "all" || currentSegement === "topics") {
+
+      return topics.map((topic: any, index: any) => (
+        <IonItem key={index} routerLink={"topics/" + topic.id}>
+          <IonThumbnail slot="start">
+            <img src={"../../assets/images/" + topic.id + ".svg"} onError={(e) => { e.currentTarget.src = "../../assets/shapes.svg" }} alt={topic.title} />
+          </IonThumbnail>
+          <IonListHeader>
+            {topic.title}
+          </IonListHeader>
+          <IonLabel>
+            {parse(topic.snippet)}
+          </IonLabel>
+        </IonItem>
+      ))
+    }
+  }
+
+  return (
+    <IonPage>
+      <IonHeader>
+        <IonToolbar>
+          <IonTitle>Search</IonTitle>
+        </IonToolbar>
+        <IonSearchbar placeholder="Find topics, cases or sections..." onIonChange={(e) => search(e)}></IonSearchbar>
```

Comment:
the `IonSegment` must be rendered here, inside IonHeader, so that it's affixed at the top and doesn't scroll away. Only render it if there is a search entered.